src/module/actors/character-sheet.ts

```diff
@@ -237,7 +262,31 @@ export class CharacterActorSheet extends ActorSheet<
           rollMode
         )
       );
+
+      // Show shock damage
+      if (game.settings.get("swnr","addShockMessage")) {
+        if (weapon.data.data.shock && weapon.data.data.shock.dmg > 0) {
+          let shock_content = `${weapon.name} Shock Damage Base ${weapon.data.data.shock.dmg} \ AC ${weapon.data.data.shock.ac}`;
+          const shockRoll = new Roll(
+             "0" +
+               " + @shockDmg + @stat.mod " +
+               (weapon.data.data.skillBoostsDamage
+                 ? ` + ${skill.data.data.rank}`
+                 : ""),
+            rollData
+          ).roll();
+          ChatMessage.create({
+            speaker: ChatMessage.getSpeaker({ actor: this.actor }),
+            flavor: shock_content,
+            type: CONST.CHAT_MESSAGE_TYPES.ROLL,
+            roll: JSON.stringify(shockRoll.toJSON()),
+          });
+        }
+      }
+
     };
+
+
```

Comment:
Roll this all into the attack roll message. Shock is part of the attack and is not a separate attack, thus should not be sent as an extra message.